File: 2013/J133.pas
var
  n, i, mode, start_y, start_w: longint;
  s: ansistring;
begin
  readln(n);
  read(s);
  setlength(s, n + 1);
  s[n + 1] := 'G';
  mode := 0;
  for i := 1 to n + 1 do
  begin
    if ((mode = 0) and (s[i] = 'Y')) then
    begin
      mode := 11;
      start_y := i
    end
    else if ((mode = 0) and (s[i] = 'W')) then
    begin
      mode := 22;
      start_w := i
    end
    else if ((mode = 11) and (s[i] = 'G')) then
    begin
      mode := 0;
      writeln('Grow ', start_y, ' ', i - 1)
    end
    else if ((mode = 12) and (s[i] = 'G')) then
    begin
      mode := 0;
      writeln('Grow ', start_w, ' ', i - 1);
      writeln('Grow ', start_y, ' ', i - 1)
    end
    else if ((mode = 21) and (s[i] = 'G')) then
    begin
      mode := 0;
      writeln('Blow ', start_y, ' ', i - 1);
      writeln('SuperGrow ', start_w, ' ', i - 1)
    end
    else if ((mode = 22) and (s[i] = 'G')) then
    begin
      mode := 0;
      writeln('SuperGrow ', start_w, ' ', i - 1)
    end
    else if ((mode = 11) and (s[i] = 'W')) then
    begin
      mode := 12;
      start_w := i
    end
    else if ((mode = 12) and (s[i] = 'Y')) then
    begin
      mode := 11;
      writeln('Grow ', start_w, ' ', i - 1)
    end
    else if ((mode = 22) and (s[i] = 'Y')) then
    begin
      mode := 21;
      start_y := i
    end
    else if ((mode = 21) and (s[i] = 'W')) then
    begin
      mode := 22;
      writeln('Blow ', start_y, ' ', i - 1)
    end
  end;
  writeln('Finish')
end.

File: 2013/J133.cpp
#include <cstdio>
char s[100001];
int main() {
  int n;
  scanf("%d", &n);
  scanf("%s", s);
  s[n] = 'G';
  char mode = 0;
  int start_y, start_w;
  for (int i = 0; i <= n; i++) {
    if (mode == 0 && s[i] == 'Y') {
      mode = 11;
      start_y = i + 1;
    } else if (mode == 0 && s[i] == 'W') {
      mode = 22;
      start_w = i + 1;
    } else if (mode == 11 && s[i] == 'G') {
      mode = 0;
      printf("Grow %d %d\n", start_y, i);
    } else if (mode == 12 && s[i] == 'G') {
      mode = 0;
      printf("Grow %d %d\n", start_w, i);
      printf("Grow %d %d\n", start_y, i);
    } else if (mode == 21 && s[i] == 'G') {
      mode = 0;
      printf("Blow %d %d\n", start_y, i);
      printf("SuperGrow %d %d\n", start_w, i);
    } else if (mode == 22 && s[i] == 'G') {
      mode = 0;
      printf("SuperGrow %d %d\n", start_w, i);
    } else if (mode == 11 && s[i] == 'W') {
      mode = 12;
      start_w = i + 1;
    } else if (mode == 12 && s[i] == 'Y') {
      mode = 11;
      printf("Grow %d %d\n", start_w, i);
    } else if (mode == 22 && s[i] == 'Y') {
      mode = 21;
      start_y = i + 1;
    } else if (mode == 21 && s[i] == 'W') {
      mode = 22;
      printf("Blow %d %d\n", start_y, i);
    }
  }
  printf("Finish\n");
  return 0;
}
